tests/Unit/ProtoTests/DeprecatedService/out/src/Client/BaseClient/DeprecatedServiceBaseClient.php

```diff
@@ -142,9 +146,27 @@ public function __construct(array $options = [])
         $this->setClientOptions($clientOptions);
     }
 
+    public function __call($method, $args)
+    {
+        if (substr($method, -5) !== 'Async') {
+            throw new ValidationException("Method name $method does not exist");
+        }
+
+        if (count($args) < 1) {
+            throw new ValidationException("Async methods require a request message");
+        }
+
+        $rpcName = substr($method, 0, -5);
+        $request = $args[0];
+        $optionalArgs = $args[1] ?? [];
+        return $this->startAsyncCall($rpcName, $request, $optionalArgs);
```

Comment:
nit: maybe this could be more succinct?
```suggestion
        return $this->startAsyncCall(substr($method, 0, -5), $args[0], $args[1] ?? []);
```

Reply:
Done.